plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hadoop/TrinoHadoopCatalog.java

```diff
@@ -299,81 +343,115 @@ public void unregisterTable(ConnectorSession session, SchemaTableName tableName)
+            CatalogUtil.dropTableData(ops.io(), ops.current());
+            trinoFileSystem.deleteDirectory(Location.of(tableLocation(schemaTableName)));
+        }
+        catch (IOException e) {
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, "Failed to delete table: " + schemaTableName, e);
+        }
+        invalidateTableCache(schemaTableName);
     }
 
     @Override
     public void dropCorruptedTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        getCatalog(session).dropTable(toTableId(schemaTableName), true);
+        dropTable(session, schemaTableName);
     }
 
     @Override
     public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
     {
-        getCatalog(session).renameTable(toTableId(from), toTableId(to));
+        throw new TrinoException(NOT_SUPPORTED, "renameTable is not supported by " + catalogName);
     }
 
     @Override
     public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
+        TableMetadata metadata;
         try {
-            return getCatalog(session).loadTable(toTableId(schemaTableName));
+            metadata = uncheckedCacheGet(
+                    tableMetadataCache,
+                    schemaTableName,
+                    () -> ((BaseTable) loadIcebergTable(this, tableOperationsProvider, session, schemaTableName)).operations().current());
         }
-        catch (NoSuchTableException e) {
-            // Have to change exception types due to code relying on specific exception to be thrown.
-            throw new TableNotFoundException(schemaTableName, e);
+        catch (UncheckedExecutionException e) {
+            throwIfUnchecked(e.getCause());
+            throw new NoSuchTableException("Table does not exist at location: %s", tableLocation(schemaTableName));
         }
-    }
 
+        return getIcebergTableWithMetadata(this, tableOperationsProvider, session, schemaTableName, metadata);
+    }
+
+//    @Override
+//    public Table loadTableWithoutCache(ConnectorSession session, SchemaTableName schemaTableName)
+//    {
+//        Table result;
+//        String location =tableLocation(schemaTableName);
+//        Pair<String, MetadataTableType> parsedMetadataType = parseMetadataType(location);
+//
+//        if (parsedMetadataType != null) {
+//            // Load a metadata table
+//            result = loadMetadataTable(session, Optional.empty(),  schemaTableName, parsedMetadataType.first(), parsedMetadataType.second());
+//        } else {
+//            // Load a normal table
+//            TableOperations ops =newTableOps(session, Optional.empty(), schemaTableName);
+//            if (ops.current() != null) {
+//                result = new BaseTable(ops, location);
+//            } else {
+//                throw new NoSuchTableException("Table does not exist at location: %s", location);
+//            }
+//        }
+//        return result;
+//    }
```

Comment:
Remove commented-out code

Reply:
I would like to keep the loadTable currently commented which is with the cache as I am pretty sure it won`t require much modification for it to work. There are other required addition for the cache to fully work but I think this method is a good start.